src/mlpack/tests/main_tests/krann_test.cpp

```diff
@@ -0,0 +1,335 @@
+  SetInputParam("k", (int) 6); // Invalid.
+
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Check that we can't specify a negative leaf size.
+ */
+BOOST_AUTO_TEST_CASE(KRANNLeafSizeTest)
+{
+  arma::mat referenceData;
+  referenceData.randu(3, 100); // 100 points in 3 dimensions.
+
+  // Random input, negative leaf size.
+  SetInputParam("reference", std::move(referenceData));
+  SetInputParam("leaf_size", (int) -1); // Invalid.
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/*
+ * Check that we can't pass both input_model and reference matrix.
+ */
+BOOST_AUTO_TEST_CASE(KRANNRefModelTest)
+{
+  arma::mat referenceData;
+  referenceData.randu(3, 100); // 100 points in 3 dimensions.
+
+  // Random input, some k <= number of  top tau %ile reference points.
+  SetInputParam("reference", std::move(referenceData));
+  SetInputParam("k", (int) 5);
+
+  mlpackMain();
+
+  // Input pre-trained model.
+  SetInputParam("input_model",
+      std::move(CLI::GetParam<RANNModel*>("output_model")));
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/*
+ * Check that we can't pass an invalid tree type.
+ */
+BOOST_AUTO_TEST_CASE(KRANNInvalidTreeTypeTest)
+{
+  arma::mat referenceData;
+  referenceData.randu(3, 100); // 100 points in 3 dimensions.
+
+  // Random input, some k <= number of  top tau %ile reference points.
+  SetInputParam("reference", std::move(referenceData));
+  SetInputParam("k", (int) 5);
+  SetInputParam("tree_type", (string) "min-rp"); // Invalid.
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+ 
+/*
+ * Check that we can't pass an invalid value of tau.
+ */
+BOOST_AUTO_TEST_CASE(KRANNInvalidTauTest)
+{
+  arma::mat referenceData;
+  referenceData.randu(3, 100); // 100 points in 3 dimensions.
+
+  // Random input, some k <= number of  top tau %ile reference points.
+  SetInputParam("reference", std::move(referenceData));
+  SetInputParam("k", (int) 5);
+  SetInputParam("tau", (double) -1); // Invalid.
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Make sure that dimensions of the neighbors and distances matrices are correct
+ * given a value of k.
+ */
+BOOST_AUTO_TEST_CASE(KRANNOutputDimensionTest)
+{
+  arma::mat referenceData;
+  referenceData.randu(3, 100); // 100 points in 3 dimensions.
+
+  // Random input, some k <= number of  top tau %ile reference points.
+  SetInputParam("reference", std::move(referenceData));
+  SetInputParam("k", (int) 5);
+
+  mlpackMain();
+
+  // Check the neighbors matrix has 5 points for each input point.
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::Mat<size_t>>
+      ("neighbors").n_rows, 5);
```

Comment:
```suggestion
  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::Mat<size_t>>("neighbors").n_rows,
      5);
```
If this line is too long, this might be a nicer-looking wrapping strategy.  If what I suggested is still too long, I'd suggest wrapping the entire `CLI::GetParam<...>` bit too. :+1: